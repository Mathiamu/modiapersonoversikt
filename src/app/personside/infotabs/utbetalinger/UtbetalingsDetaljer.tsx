import * as React from 'react';
import styled from 'styled-components';
import { Undertekst, UndertekstBold } from 'nav-frontend-typografi';
import theme from '../../../../styles/personOversiktTheme';
import { createTable, formaterNOK } from './utbetalingerUtils';
import { Ytelse, Ytelseskomponent } from '../../../../models/utbetalinger';
import { EnkeltYtelseProps } from './SammensattUtbetaling';

const Wrapper = styled.div`
  margin: .5rem 0;
  opacity: .8;
  table {
    width: 100%;
    text-align: right;
    * {
      padding: 0;
    }
  }
  th {
    font-weight: bold;
  }
  tr {
    > *:first-child {
      text-align: left;
    }
  }
`;

const Border = styled.div`
  border-top: 2px solid ${theme.color.bakgrunn};
  margin: .5rem 0;
`;

function utbetalingskomponenterListe(ytelse: Ytelse) {
    if (ytelse.ytelseskomponentListe) {
        const tittelrekke = ['Utbetalingsdetaljer', 'Sats', 'Antall', 'Beløp'];

        const tabellElementer = ytelse.ytelseskomponentListe.map((ytelseskomponent: Ytelseskomponent) => [
            ytelseskomponent.ytelseskomponenttype,
            formaterNOK(ytelseskomponent.satsbeløp || 0),
            ytelseskomponent.satsantall,
            formaterNOK(ytelseskomponent.ytelseskomponentbeløp)
        ]);

        return createTable(
            tittelrekke,
            tabellElementer
        );
    } else {
        return 'Mangler detaljer om ytelseskomponenter i utbetaling';
    }
}

function UtbetalingsDetaljer(props: EnkeltYtelseProps) {
    const ytelse = props.ytelse;
    const ytelsesKomponenter = utbetalingskomponenterListe(ytelse);
    const oversikt = createTable(
        ['Konto', 'Brutto', 'Trekk', 'Utbetalt'],
        [[
            props.konto,
            formaterNOK(ytelse.ytelseskomponentersum),
            formaterNOK(ytelse.skattsum + ytelse.trekksum),
            formaterNOK(ytelse.nettobeløp)
        ]]
    );
    return (
        <Wrapper>
            <Undertekst tag="span">
                {oversikt}
            </Undertekst>
            <Border/>
            <Undertekst tag="span">
                {ytelsesKomponenter}
            </Undertekst>
            <Border/>
            <UndertekstBold tag={'h4'}>Melding:</UndertekstBold>
            <Undertekst>{props.melding}</Undertekst>
        </Wrapper>
    );
}

export default UtbetalingsDetaljer;
