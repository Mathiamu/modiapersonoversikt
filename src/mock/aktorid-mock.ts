export function getAktorId(fodselsnummer: string): string | null {
    return `000${fodselsnummer}000`;
}
