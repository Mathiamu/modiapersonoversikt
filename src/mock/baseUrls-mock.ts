import { BaseUrlsResponse } from '../models/baseurls';

export function mockBaseUrls(): BaseUrlsResponse {
    return {
        baseUrls: [
            {
                key: 'norg2-frontend',
                url: 'https://norg2-frontend.nais.preprod.local'
            },
            {
                key: 'personforvalter',
                url: 'https://person-forvalter.nais.preprod.local/'
            },
            {
                key: 'drek',
                url: 'https://drek.nais.preprod.local/'
            }
        ]
    };
}
