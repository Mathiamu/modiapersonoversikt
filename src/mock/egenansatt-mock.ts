import * as faker from 'faker/locale/nb_NO';
import { vektetSjanse } from './utils/mock-utils';
import { Egenansatt } from '../models/egenansatt';

export function erEgenAnsatt(f√∏dselsnummer: String): Egenansatt {
    return { erEgenAnsatt : getEgenAnsattMarkering() };
}

function getEgenAnsattMarkering(): boolean {
    if (vektetSjanse(faker, 0.9)) {
        return true;
    } else {
        return false;
    }
}