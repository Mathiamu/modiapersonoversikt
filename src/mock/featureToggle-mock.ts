import { FeatureToggleResponse } from '../models/featureToggle';
import { FeatureToggles } from '../components/featureToggle/toggleIDs';

export function mockFeatureToggle(toggleId: FeatureToggles): FeatureToggleResponse {
    switch (toggleId) {
        case FeatureToggles.VisTilbakemelding:
            return true;
        case FeatureToggles.SaksDokumentIEgetVindu:
            return true;
        case FeatureToggles.SvaksyntModus:
            return false;
        case FeatureToggles.AvsluttGosysOppgave:
            return true;
        default:
            return Math.random() > 0.5;
    }
}
