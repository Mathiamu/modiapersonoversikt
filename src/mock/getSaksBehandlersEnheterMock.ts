import { enheter } from './context-mock';
import { SaksbehandlersEnheter } from '../rest/resources/saksbehandlersEnheter';

export function getSaksBehandlersEnheterMock(): SaksbehandlersEnheter {
    return {
        ident: 'Z990000',
        enhetliste: enheter
    };
}
