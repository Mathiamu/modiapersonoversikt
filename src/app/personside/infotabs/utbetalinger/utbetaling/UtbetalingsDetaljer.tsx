import * as React from 'react';
import styled from 'styled-components';
import { Undertekst, UndertekstBold } from 'nav-frontend-typografi';
import theme from '../../../../../styles/personOversiktTheme';
import {
    createTable,
    formaterNOK,
    skattBelopAscComparator, trekkBelopAscComparator,
    ytelseBelopDescComparator
} from '../utils/utbetalingerUtils';
import { Ytelse, Ytelseskomponent } from '../../../../../models/utbetalinger';
import { EnkeltYtelseProps } from './SammensattUtbetaling';
import AlertStripeAdvarsel from 'nav-frontend-alertstriper/lib/advarsel-alertstripe';

const Wrapper = styled.aside`
  padding: .5rem 0;
  th {
    font-weight: bold;
  }
`;

const Border = styled.div`
  border-top: 2px solid ${theme.color.bakgrunn};
  margin: .5rem 0;
`;

function utbetalingsDetaljerTable(ytelse: Ytelse) {
    if (ytelse.ytelseskomponentListe || ytelse.trekkListe || ytelse.skattListe) {
        const tittelrekke = ['Detaljer', 'Sats', 'Antall', 'Beløp'];
        let tabellElementer: Array<Array<number | string | undefined>> = [];

        if (ytelse.ytelseskomponentListe) {
            ytelse.ytelseskomponentListe
                .sort(ytelseBelopDescComparator)
                .map((ytelseskomponent: Ytelseskomponent) =>
                    tabellElementer.push([
                        ytelseskomponent.ytelseskomponenttype,
                        ytelseskomponent.satsbeløp ? formaterNOK(ytelseskomponent.satsbeløp) : '',
                        ytelseskomponent.satsantall,
                        formaterNOK(ytelseskomponent.ytelseskomponentbeløp)
            ]));
        }

        if (ytelse.skattListe) {
            ytelse.skattListe
                .sort(skattBelopAscComparator)
                .map(skattElement =>
                    tabellElementer.push([
                        'Skattetrekk',
                        '',
                        '',
                        formaterNOK(skattElement.skattebeløp)
            ]));
        }

        if (ytelse.trekkListe) {
            ytelse.trekkListe
                .sort(trekkBelopAscComparator)
                .map(trekkElement =>
                    tabellElementer.push([
                        trekkElement.trekktype,
                        '',
                        '',
                        formaterNOK(trekkElement.trekkbeløp)
            ]));
        }

        return createTable(
            tittelrekke,
            tabellElementer
        );
    } else {
        return <AlertStripeAdvarsel>Manglende data. Kunne ikke finne detaljer om utbetaling.</AlertStripeAdvarsel>;
    }
}

function UtbetalingsDetaljer(props: EnkeltYtelseProps) {
    const ytelse = props.ytelse;
    const ytelsesKomponenter = utbetalingsDetaljerTable(ytelse);
    const oversikt = createTable(
        ['Konto', 'Brutto', 'Trekk', 'Utbetalt'],
        [[
            props.konto,
            formaterNOK(ytelse.ytelseskomponentersum),
            formaterNOK(ytelse.skattsum + ytelse.trekksum),
            formaterNOK(ytelse.nettobeløp)
        ]]
    );
    return (
        <Wrapper>
            <Undertekst tag="span">
                {oversikt}
            </Undertekst>
            <Border/>
            <Undertekst tag="span">
                {ytelsesKomponenter}
            </Undertekst>
            <Border/>
            <UndertekstBold tag={'h4'}>Melding</UndertekstBold>
            <Undertekst>{props.melding || ''}</Undertekst>
        </Wrapper>
    );
}

export default UtbetalingsDetaljer;
