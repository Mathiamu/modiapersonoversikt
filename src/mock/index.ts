import faker from 'faker/locale/nb_NO';
import navfaker from 'nav-faker';
import Cookies from 'js-cookie';
import FetchMock, { Middleware, MiddlewareUtils, MockHandler, MockRequest } from 'yet-another-fetch-mock';
import { erGyldigFødselsnummer } from 'nav-faker/dist/personidentifikator/helpers/fodselsnummer-utils';
import { apiBaseUri } from '../api/config';
import { getPerson } from './person/personMock';
import { getMockKontaktinformasjon } from './person/krrKontaktinformasjon/kontaktinformasjon-mock';
import {
    mockGeneratorMedEnhetId,
    mockGeneratorMedFodselsnummer,
    verify,
    withDelayedResponse
} from './utils/fetch-utils';
import { getMockNavKontor } from './navkontor-mock';
import { erEgenAnsatt } from './egenansatt-mock';
import { mockBaseUrls } from './baseUrls-mock';
import { getMockVeilederRoller } from './veilderRoller-mock';
import { mockRetningsnummereKodeverk } from './kodeverk/retningsnummer-mock';
import { mockTilrettelagtKommunikasjonKodeverk } from './kodeverk/tilrettelagt-kommunikasjon-kodeverk-mock';
import { mockPostnummere } from './kodeverk/postnummer-kodeverk-mock';
import { mockLandKodeverk } from './kodeverk/land-kodeverk-mock';
import { mockValutaKodeverk } from './kodeverk/valuta-kodeverk-mock';
import { mockVergemal } from './person/vergemal/vergemalMock';
import { getMockUtbetalinger } from './utbetalinger/utbetalinger-mock';
import { getMockSykepengerRespons } from './ytelse/sykepenger-mock';
import { getMockForeldrepenger } from './ytelse/foreldrepenger-mock';
import { getMockPleiepenger } from './ytelse/pleiepenger-mock';
import { mockFeatureToggle } from './featureToggle-mock';
import { getMockSaksoversikt } from './saksoversikt/saksoversikt-mock';
import { getMockOppfolging, getMockYtelserOgKontrakter } from './oppfolging-mock';
import { getDittNavVarsler, getMockVarsler } from './varsler/varsel-mock';
import { getForeslattEnhet, getMockAnsatte, getMockEnheter, getMockGsakTema } from './meldinger/oppgave-mock';
import { getMockInnloggetSaksbehandler } from './innloggetSaksbehandler-mock';
import { saker } from './journalforing/journalforing-mock';
import { mockPersonsokResponse, mockStaticPersonsokRequest } from './person/personsokMock';
import { setupWsControlAndMock } from './context-mock';
import standardTekster from './standardTeksterMock.js';
import { getSaksBehandlersEnheterMock } from './getSaksBehandlersEnheterMock';
import { saksbehandlerCookieNavnPrefix } from '../redux/session/saksbehandlersEnhetCookieUtils';
import { OppgaverBackendMock } from './mockBackend/oppgaverBackendMock';
import { setupSaksbehandlerInnstillingerMock } from './saksbehandlerinnstillinger-mock';
import { failurerateMiddleware } from './utils/failureMiddleware';
import { setupDraftMock } from './draft-mock';
import { authMock, tilgangskontrollMock } from './tilgangskontroll-mock';
import { delayed } from './utils-mock';
import { MeldingerBackendMock } from './mockBackend/meldingerBackendMock';
import { setupSFDialogMock } from './dialoger/sf-dialoger-mock';
import { setupHenvendelseDialogMock } from './dialoger/henvendelse-dialoger-mock';
import { FeatureToggles } from '../components/featureToggle/toggleIDs';
import { getAktorId } from './aktorid-mock';
import { setupPersondataMock } from './persondata/persondata';

const STATUS_OK = () => 200;
const STATUS_BAD_REQUEST = () => 400;

const oppgaveBackendMock = new OppgaverBackendMock();
const meldingerBackendMock = new MeldingerBackendMock(oppgaveBackendMock);

const harEnhetIdSomQueryParam = (req: MockRequest) => {
    const enhetQueryParam = req.queryParams.enhet;
    if (!enhetQueryParam) {
        return 'Skal ha enhetId i queryParameter';
    }
    return undefined;
};

export function randomDelay() {
    if (navfaker.random.vektetSjanse(0.05)) {
        return faker.random.number(5000);
    }
    return faker.random.number(750);
}

const fodselsNummerErGyldigStatus = (req: MockRequest) =>
    erGyldigFødselsnummer(req.pathParams.fodselsnummer) ? STATUS_OK() : STATUS_BAD_REQUEST();

function setupInnloggetSaksbehandlerMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/hode/me',
        withDelayedResponse(randomDelay(), STATUS_OK, () => getMockInnloggetSaksbehandler())
    );
}

function setUpSaksbehandlersEnheterMock(mock: FetchMock) {
    mock.get(apiBaseUri + '/hode/enheter', withDelayedResponse(randomDelay(), STATUS_OK, getSaksBehandlersEnheterMock));
}

function setupTilgangskontroll(mock: FetchMock) {
    mock.get(
        '/modiapersonoversikt-api/rest/tilgang/auth',
        withDelayedResponse(randomDelay(), () => 200, authMock)
    );

    mock.get(
        '/modiapersonoversikt-api/rest/tilgang/:fodselsnummer?',
        withDelayedResponse(
            randomDelay(),
            () => (Math.random() > 0.98 ? 400 : 200),
            mockGeneratorMedFodselsnummer(tilgangskontrollMock)
        )
    );
}

function setupPersonMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/person/:fodselsnummer',
        withDelayedResponse(
            randomDelay(),
            fodselsNummerErGyldigStatus,
            mockGeneratorMedFodselsnummer(fodselsnummer => getPerson(fodselsnummer))
        )
    );
}

function setupAktorIdMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/person/:fodselsnummer/aktorid',
        withDelayedResponse(
            randomDelay(),
            fodselsNummerErGyldigStatus,
            mockGeneratorMedFodselsnummer(fodselsnummer => getAktorId(fodselsnummer))
        )
    );
}

function setupEgenAnsattMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/egenansatt/:fodselsnummer',
        withDelayedResponse(
            randomDelay(),
            fodselsNummerErGyldigStatus,
            mockGeneratorMedFodselsnummer(fodselsnummer => erEgenAnsatt(fodselsnummer))
        )
    );
}

function setupKontaktinformasjonMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/person/:fodselsnummer/kontaktinformasjon',
        withDelayedResponse(
            randomDelay(),
            fodselsNummerErGyldigStatus,
            mockGeneratorMedFodselsnummer(fodselsnummer => getMockKontaktinformasjon(fodselsnummer))
        )
    );
}

function setupSaksoversiktMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/saker/:fodselsnummer/sakstema',
        verify(
            harEnhetIdSomQueryParam,
            withDelayedResponse(
                randomDelay(),
                fodselsNummerErGyldigStatus,
                mockGeneratorMedFodselsnummer(getMockSaksoversikt)
            )
        )
    );
}

function setupUtbetalingerMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/utbetaling/:fodselsnummer',
        withDelayedResponse(randomDelay(), fodselsNummerErGyldigStatus, args =>
            getMockUtbetalinger(args.pathParams.fodselsnummer, args.queryParams.startDato, args.queryParams.sluttDato)
        )
    );
}

function setupSykepengerMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/ytelse/sykepenger/:fodselsnummer',
        withDelayedResponse(
            randomDelay(),
            fodselsNummerErGyldigStatus,
            mockGeneratorMedFodselsnummer(fodselsnummer => getMockSykepengerRespons(fodselsnummer))
        )
    );
}

function setupForeldrepengerMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/ytelse/foreldrepenger/:fodselsnummer',
        withDelayedResponse(
            randomDelay(),
            fodselsNummerErGyldigStatus,
            mockGeneratorMedFodselsnummer(fodselsnummer => getMockForeldrepenger(fodselsnummer))
        )
    );
}

function setupPleiepengerMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/ytelse/pleiepenger/:fodselsnummer',
        withDelayedResponse(
            randomDelay(),
            fodselsNummerErGyldigStatus,
            mockGeneratorMedFodselsnummer(fodselsnummer => getMockPleiepenger(fodselsnummer))
        )
    );
}

function setupOppfolgingMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/oppfolging/:fodselsnummer',
        withDelayedResponse(
            randomDelay(),
            fodselsNummerErGyldigStatus,
            mockGeneratorMedFodselsnummer(fodselsnummer => getMockOppfolging(fodselsnummer))
        )
    );
}

function setupYtelserOgKontrakter(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/oppfolging/:fodselsnummer/ytelserogkontrakter',
        withDelayedResponse(
            randomDelay(),
            fodselsNummerErGyldigStatus,
            mockGeneratorMedFodselsnummer(fodselsnummer => getMockYtelserOgKontrakter(fodselsnummer))
        )
    );
}

function setupVarselMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/varsler/:fodselsnummer',
        withDelayedResponse(
            randomDelay(),
            fodselsNummerErGyldigStatus,
            mockGeneratorMedFodselsnummer(fodselsnummer => getMockVarsler(fodselsnummer))
        )
    );

    const dittnaveventHandler: MockHandler = (req, res, ctx) => {
        const headers: any = req.init?.headers;
        const fnr = headers.fodselsnummer;
        if (!erGyldigFødselsnummer(fnr)) {
            return res(ctx.status(400));
        }
        return res(ctx.status(200), ctx.json(getDittNavVarsler(fnr)));
    };

    mock.get(
        '/modiapersonoversikt/proxy/dittnav-eventer-modia/fetch/beskjed/all',
        delayed(randomDelay(), dittnaveventHandler)
    );
    mock.get(
        '/modiapersonoversikt/proxy/dittnav-eventer-modia/fetch/oppgave/all',
        delayed(randomDelay(), dittnaveventHandler)
    );
}

function setupGsakTemaMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/dialogoppgave/tema',
        withDelayedResponse(randomDelay(), STATUS_OK, () => getMockGsakTema())
    );
}

function setupOppgaveEnhetMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/enheter/oppgavebehandlere/alle',
        withDelayedResponse(randomDelay(), STATUS_OK, () => getMockEnheter())
    );
}

function setupForeslatteEnheterMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/enheter/oppgavebehandlere/foreslatte',
        withDelayedResponse(randomDelay(), STATUS_OK, () => getForeslattEnhet())
    );
}

function setupAnsattePaaEnhetMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/enheter/:enhetId/ansatte',
        withDelayedResponse(
            randomDelay(),
            STATUS_OK,
            mockGeneratorMedEnhetId(enhetId => getMockAnsatte(enhetId))
        )
    );
}

function setupGeografiskTilknytningMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/enheter',
        withDelayedResponse(
            randomDelay(),
            (args: MockRequest) => {
                if (isNaN(args.queryParams.gt) && !args.queryParams.dkode) {
                    return 404;
                } else {
                    return 200;
                }
            },
            (args: MockRequest) => getMockNavKontor(args.queryParams.gt, args.queryParams.dkode)
        )
    );
}

function setupPersonsokMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/personsok',
        withDelayedResponse(randomDelay(), STATUS_OK, () => mockPersonsokResponse(mockStaticPersonsokRequest()))
    );
}

function setSaksbehandlerinnstillingerMockBackend(args: MockRequest) {
    Cookies.set(saksbehandlerCookieNavnPrefix + '-Z990099', args.body);
    return args.body;
}

function setupVelgEnhetMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/hode/velgenhet',
        withDelayedResponse(randomDelay(), STATUS_OK, setSaksbehandlerinnstillingerMockBackend)
    );
}

function setupOppgaveMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/oppgaver/plukk/:temagruppe',
        verify(
            harEnhetIdSomQueryParam,
            withDelayedResponse(randomDelay(), STATUS_OK, () => oppgaveBackendMock.plukkOppgave())
        )
    );
}

function setupTildelteOppgaverMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/oppgaver/tildelt/:fnr',
        withDelayedResponse(randomDelay(), STATUS_OK, () => oppgaveBackendMock.getTildelteOppgaver())
    );
}

function setupLeggTilbakeOppgaveMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/oppgaver/legg-tilbake',
        withDelayedResponse(randomDelay(), STATUS_OK, request => oppgaveBackendMock.leggTilbake(request.body))
    );
}

function setupVergemalMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/person/:fodselsnummer/vergemal',
        withDelayedResponse(
            randomDelay(),
            fodselsNummerErGyldigStatus,
            mockGeneratorMedFodselsnummer(fodselsnummer => mockVergemal(fodselsnummer))
        )
    );
}

function setupBaseUrlsMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/baseurls',
        withDelayedResponse(randomDelay(), STATUS_OK, () => {
            return mockBaseUrls();
        })
    );
}

function setupFeatureToggleMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/featuretoggle/:toggleId',
        withDelayedResponse(
            randomDelay(),
            STATUS_OK,
            // @ts-ignore
            (args: HandlerArgument) => mockFeatureToggle(args.pathParams.toggleId)
        )
    );
}

function setupVeilederRollerMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/veileder/roller',
        withDelayedResponse(randomDelay(), STATUS_OK, () => getMockVeilederRoller())
    );
}

function setupRetningsnummerKodeverkMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/kodeverk/Retningsnumre',
        withDelayedResponse(randomDelay(), STATUS_OK, () => mockRetningsnummereKodeverk())
    );
}

function setupTilrettelagtKommunikasjonKodeverkMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/kodeverk/TilrettelagtKommunikasjon',
        withDelayedResponse(randomDelay(), STATUS_OK, () => mockTilrettelagtKommunikasjonKodeverk())
    );
}

function setupPostnummerKodeverk(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/kodeverk/Postnummer',
        withDelayedResponse(randomDelay(), STATUS_OK, () => mockPostnummere())
    );
}

function setupLandKodeverk(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/kodeverk/Landkoder',
        withDelayedResponse(randomDelay(), STATUS_OK, () => mockLandKodeverk())
    );
}

function setupValutaKodeverk(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/kodeverk/Valutaer',
        withDelayedResponse(randomDelay(), STATUS_OK, () => mockValutaKodeverk())
    );
}

function setupJournalforingMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/journalforing/:fnr/saker/',
        withDelayedResponse(randomDelay(), STATUS_OK, () => saker)
    );
    mock.post(
        apiBaseUri + '/journalforing/:fnr/:traadId',
        verify(
            harEnhetIdSomQueryParam,
            withDelayedResponse(randomDelay(), STATUS_OK, () => ({}))
        )
    );
}

function opprettOppgaveMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/dialogoppgave/opprett',
        withDelayedResponse(randomDelay(), STATUS_OK, () => ({}))
    );
}

function opprettSkjermetOppgaveMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/dialogoppgave/opprettskjermetoppgave',
        withDelayedResponse(randomDelay(), STATUS_OK, () => ({}))
    );
}

function setupStandardteksterMock(mock: FetchMock) {
    mock.get(
        '/modiapersonoversikt-skrivestotte/skrivestotte',
        withDelayedResponse(randomDelay(), STATUS_OK, () => standardTekster)
    );

    mock.post(
        '/modiapersonoversikt-skrivestotte/skrivestotte/statistikk/:id',
        withDelayedResponse(randomDelay(), STATUS_OK, () => undefined)
    );
}

const contentTypeMiddleware: Middleware = (requestArgs, response) => {
    if (response.headers) {
        return response;
    }
    response.headers = {
        'content-type': 'application/json'
    };
    return response;
};

console.log('=========================='); // tslint:disable-line
console.log('======== MED MOCK ========'); // tslint:disable-line
console.log('=========================='); // tslint:disable-line

const mock = FetchMock.configure({
    enableFallback: true,
    middleware: MiddlewareUtils.combine(
        contentTypeMiddleware,
        failurerateMiddleware(0.02),
        MiddlewareUtils.loggingMiddleware()
    )
});

setupInnloggetSaksbehandlerMock(mock);
setupPersonMock(mock);
setupPersondataMock(mock);
setupTilgangskontroll(mock);
setupEgenAnsattMock(mock);
setupKontaktinformasjonMock(mock);
setupGeografiskTilknytningMock(mock);
setupSaksoversiktMock(mock);
setupUtbetalingerMock(mock);
setupSykepengerMock(mock);
setupForeldrepengerMock(mock);
setupPleiepengerMock(mock);
setupOppgaveMock(mock);
setupAktorIdMock(mock);

if (mockFeatureToggle(FeatureToggles.BrukSalesforceDialoger)) {
    setupSFDialogMock(mock, meldingerBackendMock);
} else {
    setupHenvendelseDialogMock(mock, meldingerBackendMock);
}

setupTildelteOppgaverMock(mock);
setupLeggTilbakeOppgaveMock(mock);
setupVergemalMock(mock);
setupBaseUrlsMock(mock);
setupFeatureToggleMock(mock);
setupVeilederRollerMock(mock);
setupRetningsnummerKodeverkMock(mock);
setupTilrettelagtKommunikasjonKodeverkMock(mock);
setupPostnummerKodeverk(mock);
setupWsControlAndMock(mock);
setupLandKodeverk(mock);
setupValutaKodeverk(mock);
setupOppfolgingMock(mock);
setupGsakTemaMock(mock);
setupOppgaveEnhetMock(mock);
setupForeslatteEnheterMock(mock);
setupAnsattePaaEnhetMock(mock);
setupYtelserOgKontrakter(mock);
setupVarselMock(mock);
opprettOppgaveMock(mock);
opprettSkjermetOppgaveMock(mock);
setupPersonsokMock(mock);
setupJournalforingMock(mock);
setupStandardteksterMock(mock);
setupVelgEnhetMock(mock);
setUpSaksbehandlersEnheterMock(mock);
setupSaksbehandlerInnstillingerMock(mock);
setupDraftMock(mock);
