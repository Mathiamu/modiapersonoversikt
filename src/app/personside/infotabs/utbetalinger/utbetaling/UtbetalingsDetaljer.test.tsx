import * as React from 'react';
import * as renderer from 'react-test-renderer';
import UtbetalingsDetaljer from './UtbetalingsDetaljer';
import { statiskMockYtelse } from '../../../../../mock/statiskMockUtbetaling';

test('Viser utbetalingsdetaljer riktig med liste med ytelser og trekk', () => {
    const visittkortheader = renderer.create(
        <UtbetalingsDetaljer
            konto="123"
            melding="Dette er en melding"
            ytelse={statiskMockYtelse}
            open={true}
            toggleVisDetaljer={() => null}
        />
    );

    const json = visittkortheader.toJSON();
    expect(json).toMatchSnapshot();
});
