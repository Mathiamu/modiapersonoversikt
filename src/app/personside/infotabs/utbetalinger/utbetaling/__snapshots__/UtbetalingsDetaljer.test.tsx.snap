// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Viser utbetalingsdetaljer riktig med liste med ytelser og trekk 1`] = `
.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c4 {
  font-weight: bold;
}

.c0 {
  -webkit-transition: .5s;
  transition: .5s;
  padding: 0 1.25rem 1.25rem;
  border-radius: 0.25rem;
  background-color: #f4f4f4;
  box-shadow: inset 0 0 0 1px #59514b;
  padding: 1.25rem;
  margin: 1.25rem 0 0;
}

.c6 {
  border: none;
  padding: .1rem .2rem;
  margin-top: 1rem;
  border-radius: .35rem;
  cursor: pointer;
  background-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #0067c5;
}

.c6:focus {
  outline: none;
  box-shadow: 0 0 0 0.1875rem #FFBD66;
}

.c6:hover {
  opacity: 0.8;
}

.c7 {
  margin-right: .5rem;
}

.c1 th:not(:first-child) {
  font-weight: normal;
  font-style: italic;
}

.c2 td:not(:first-child) {
  font-weight: bold;
}

.c3 {
  border-top: 2px solid #efefef;
  margin: .5rem 0;
}

<div
  className="c0"
  open={true}
>
  <aside
    className="c1"
  >
    <span
      className="typo-normal"
    >
      <div
        className="c2"
      >
        <table>
          <thead>
            <tr>
              <th>
                Konto
              </th>
              <th>
                Brutto
              </th>
              <th>
                Trekk
              </th>
              <th>
                Utbetalt
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                123
              </td>
              <td>
                13,370.00
              </td>
              <td>
                -1,984.00
              </td>
              <td>
                11,386.00
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </span>
    <div
      className="c3"
    />
    <span
      className="typo-normal"
    >
      <table>
        <thead>
          <tr>
            <th>
              Detaljer
            </th>
            <th>
              Sats
            </th>
            <th>
              Antall
            </th>
            <th>
              Bel√∏p
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              Bacon
            </td>
            <td>
              558.00
            </td>
            <td>
              10
            </td>
            <td>
              5,580.00
            </td>
          </tr>
          <tr>
            <td>
              Computer
            </td>
            <td>
              1,310.00
            </td>
            <td>
              3
            </td>
            <td>
              3,930.00
            </td>
          </tr>
          <tr>
            <td>
              Mouse
            </td>
            <td>
              1,930.00
            </td>
            <td>
              2
            </td>
            <td>
              3,860.00
            </td>
          </tr>
          <tr>
            <td>
              Skattetrekk
            </td>
            <td>
              
            </td>
            <td>
              
            </td>
            <td>
              -730.00
            </td>
          </tr>
          <tr>
            <td>
              Prosenttrekk
            </td>
            <td>
              
            </td>
            <td>
              
            </td>
            <td>
              -1,042.00
            </td>
          </tr>
          <tr>
            <td>
              Prosenttrekk
            </td>
            <td>
              
            </td>
            <td>
              
            </td>
            <td>
              -212.00
            </td>
          </tr>
        </tbody>
      </table>
    </span>
    <div
      className="c3"
    />
    <h4
      className="typo-normal"
    >
      <span
        className="c4"
      >
        Melding
      </span>
    </h4>
    <p
      className="typo-normal"
    >
      Dette er en melding
    </p>
  </aside>
  <div
    className="c5"
  >
    <button
      aria-expanded={true}
      className="c6"
      onClick={[Function]}
      open={true}
    >
      <span
        className="c7"
      >
        <span
          className="typo-normal"
        >
          Skjul
           detaljer
        </span>
      </span>
      <i
        className="nav-frontend-chevron chevronboks chevron--opp"
      />
    </button>
  </div>
</div>
`;
